src/lib/wallet/GoodWalletProvider.js

```diff
@@ -52,6 +52,10 @@ export const GoodWalletProvider = ({ children, disableLoginAndWatch = false }) =
 
   const initWalletAndStorage = useCallback(
     async (seedOrWeb3, type: 'SEED' | 'METAMASK' | 'WALLETCONNECT' | 'OTHER') => {
+      if (goodWallet && userStorage) {
```

Comment:
@roma-hladilka please explain the fix

Reply:
reinitialization make storage work incorrect, for instance, userStorage.getPrivateProfile returns an empty object. So it is just a check to not reinitialize it

Reply:
@roma-hladilka 
reinitialization is a part of logic.
firstly storage initializes for undefined private key (no user), then it reinitializes after torus login with the private key we got

i'm reverting this. adding more exact deps for the hooks & effects
pls re-pull branch and continue research - why reinitialization doesn't works as expected

Reply:
@johnsmith-gooddollar @roma-hladilka 
i've solved this issue. problem was that initRegistered() once called on subsequent initWalletAndStorage
- i've moved the call to initwallet into appswitch
- made sure initialized is only called once (when ready is false)
- added a flag to initWalletAndStorage to also called initRegistered (ie required when user is logged in)